### ISSUE TITLE
username/password saved as cleartext when using URIs with user:pass@ format

### ISSUE THREAD

**Issue by curiositycasualty** [saltstack/salt#38187]
### Description of Issue/Question
When doing something like `__salt__['cp.cache_file']('user:pass@google.com:8081/sub/path/file.ext')` you end up with:
```tree
[vagrant@vm ~]$ tree /var/cache/salt/minion/extrn_files/
/var/cache/salt/minion/extrn_files/
└── base
    ├── user:pass@google.com:8081
    │   └── sub
    │       └── path
    │            └── file.ext
  ...
```

### Steps to Reproduce Issue
`sudo salt-call cp.cache_file 'user:pass@google.com:8081/sub/path/file.ext' -l debug`

### Versions Report
```
[vagrant@vm ~]$ sudo salt-call --versions-report
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
   python-gnupg: 0.3.8
         PyYAML: 3.10
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.22.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

**Comment by gtmanfred** [comment]
We should clean that up.

Thanks for reporting this,
Daniel

**Comment by techhat** [comment]
Fixed by #38541.

**Comment by cachedout** [comment]
Closed via #38541
